File: code-examples/scala/authentication/sha256.scala
import org.mongodb.scala._

object MongoConnect {
  def main(args: Array[String]): Unit = {
    // Replace the following with values for your environment.
    val username = "<username>"
    val password = "<password>"
    val clusterUrl = "<MongoDB cluster url>"

    val authMechanism = "SCRAM-SHA-256"

    // Replace the following with your MongoDB deployment's connection string.
    val uri =
      s"mongodb+srv://${username}:${password}@${clusterUrl}/?authMechanism=${authMechanism}"

    // Create a new MongoClient
    val client = MongoClient(uri)
    
    // Function to connect to the server
    def run(): Unit = {
      client.getDatabase("admin").runCommand(Document("ping" -> 1)).collect().subscribe((result: Seq[Document]) => {
        println("Connected successfully to server")
      }, (e: Throwable) => e.printStackTrace(), () => client.close())
    }
    run()
  }
}

File: code-examples/php/authentication/sha256.php
<?php

require_once __DIR__ . "/vendor/autoload.php";

// Replace the following with your own values.
$username = "<username>";
$password = "<password>";
$clusterUrl = "<MongoDB cluster url>";

$authMechanism = "SCRAM-SHA-256";

// Replace the following with your MongoDB deployment's connection string.
$uri = sprintf('mongodb+srv://%s:%s@%s/?authMechanism=%s', 
    urlencode($username), urlencode($password), $clusterUrl, $authMechanism);

// Create a new MongoDB client
$client = new MongoDB\Client($uri);

try {
    // Select the database and run the ping command
    $command = new MongoDB\Driver\Command(['ping' => 1]);
    $client->getManager()->executeCommand('admin', $command);
    
    // If no exception thrown, we connected successfully.
    echo "Connected successfully to server\n";
    
} catch(MongoDB\Driver\Exception\Exception $e) {
    echo $e->getMessage(), "\n";
} finally {
    // Close the connection
    $client = null;
}
?>
